src/main/java/com/melon/portfoliomanager/services/CompanyStocksManager.java

```diff
@@ -2,10 +2,34 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 public class CompanyStocksManager {
+    private final double PRICE_CHANGE_PERCENT_THRESHOLD = 5;
     private final Map<String, Double> companyStocks = new HashMap<>();
 
+    public Map<String, Double> getCompanyStocksWithExtremePriceChanges(Map<String, Double> latestCompanyStocks) {
+        return companyStocks.entrySet()
+                .stream()
+                .filter(entry -> {
+                    double latestStockPrice = latestCompanyStocks.get(entry.getKey());
+                    double lastStockPrice = entry.getValue();
+                    return isPriceChangeAboveThreshold(latestStockPrice, lastStockPrice);
+                })
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private boolean isPriceChangeAboveThreshold(double latestStockPrice, double lastStockPrice) {
```

Comment:
You can turn this method into a single line:

return Math.abs(((latestStockPrice - lastStockPrice) / lastStockPrice) * 100) > PRICE_CHANGE_PERCENT_THRESHOLD;